Orange/widgets/data/owneighbors.py

```diff
@@ -88,24 +82,53 @@ def __init__(self):
 
         self.apply_button = gui.auto_apply(self.controlArea, self, commit=self.apply)
 
-    def _set_label_text(self, name):
-        data = getattr(self, name)
-        label = getattr(self, f"{name}_info_label")
-        if data is None:
-            label.setText(f"No {name} instances")
+    def _set_input_summary(self):
+        n_data, n_refs = 0, 0
+        if self.data is not None and self.reference is not None:
+            n_data = len(self.data)
+            n_refs = len(self.reference)
+        elif self.data is not None:
+            n_data = len(self.data)
+        elif self.reference is not None:
+            n_refs = len(self.reference)
```

Comment:
You can shorten this by:

```
n_data = len(self.data) if self.data else 0
n_refs = len(self.reference) if self.reference else 0
```

and move the following `else` statement into the second `if`